secretsmanager/src/secrets_manager_enc.cpp

```diff
@@ -0,0 +1,81 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+//
+// This convenience header combines different block implementations.
+
+#include "secrets_manager_enc.h"
+
+#include "aes.h"
+#include "openssl_pass.h"
+#include "base64.h"
+
+namespace concord::secretsmanager {
+
+SecretsManagerEnc::SecretsManagerEnc(SecretRetriever::SecretData& secrets) {
+  if (supported_digets_.find(secrets.digest) == supported_digets_.end()) {
```

Comment:
typo: `supported_digets` -> `supported_digests`

Reply:
This contains some custom steps - Salted__ and the salt itself is prepended to the cipher text. That's why it is in src folder.